File: anarc08h.cpp
#include <bits/stdc++.h>
#include <cstdio>
#include <sstream>
using namespace std;

int main(){
    int n,d;
    scanf("%d %d", &n, &d);
    while (n && d){
    vector<int> chairs(n);
    iota(chairs.begin(), chairs.end(),1);

    while (chairs.size()!=1){
        int size = chairs.size();
        if (d>(size-1)){
            chairs.erase(chairs.begin() + ((d-1)%size));
        }
        else{
            std::vector<int> first(chairs.begin(),chairs.begin()+(d-1));
            std::vector<int> second(chairs.begin()+d,chairs.end());

            chairs = second;
            chairs.insert(chairs.end(),first.begin(),first.end());
        }
        
    }
    printf("%d %d %d\n",n,d,chairs[0] );
    scanf("%d %d", &n, &d);
    }
    return 0;
    //What about sliding window algorithm? 
    //I could split and merge vectors though
}

File: anarc08h.py
while True:
    x = raw_input("").split()
    n, d = int(x[0]), int(x[1])-1
    if not n:
        break
    chairs = list(range(1,n+1))
    while (len(chairs)!=1):
        if d>len(chairs)-1:
            del chairs[d%len(chairs)]
            print chairs
        else:
            first = chairs[d+1::] #4,5
            second = chairs[:d:] 
            del chairs[d]
            chairs = first + second
            print chairs
    print n, d+1, chairs[0]

